### PR TITLE
Refactor ExhaustiveDeps

### PR DESCRIPTION
**Pull request by its-mXc**
## Summary
Destructure, use includes instead on indexOf === -1, and remove unused variables in arguments

### CODE REVIEW

**Review by SMAKSS — COMMENTED** [review]
Hi there, can you tell us why these changes, make the current code base better? Does it affect performance or it provides more readability? or something else?

**Comment by its-mXc** [comment]
@SMAKSS This doesn't affect performance, just makes it a bit more readable

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 1]
```diff
@@ -1376,7 +1376,7 @@ function collectRecommendations({
   declaredDependencies.forEach(({key}) => {
     // Does this declared dep satisfy a real need?
     if (satisfyingDependencies.has(key)) {
-      if (suggestedDependencies.indexOf(key) === -1) {
+      if (!suggestedDependencies.includes(key)) {
```
Comment (robertpiosik) [inline packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:1379]: [`Array.prototype.includes()` is not supported in IE.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 2]
```diff
@@ -1653,7 +1653,7 @@ function analyzePropertyChain(node, optionalChains) {
   }
 }
 
-function getNodeWithoutReactNamespace(node, options) {
+function getNodeWithoutReactNamespace(node) {
```
Comment (robertpiosik) [inline packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:1656]: This is actually the only little thing I would agree in this PR, there should be not unused params indeed.

**Review by robertpiosik — COMMENTED** [review]
In my opinion syntax features like destructuring assignment are always optional and equivalent here.

They are convenient in many cases, especially when you are pulling out multiple things and you can do it in a single statement then.

Code here always pull out single elements from objects so I would say it's not necessary to refactor it.

**Comment by its-mXc** [comment]
@robertpiosik Thanks for the feedback. I will close this

**Pull request closed without merging**